src/System.IO.Compression/src/System/IO/Compression/DeflateStream.cs

```diff
@@ -202,14 +202,26 @@ public override long Position
         public override void Flush()
         {
             EnsureNotDisposed();
+            FlushBuffers();
         }
 
         public override Task FlushAsync(CancellationToken cancellationToken)
         {
             EnsureNotDisposed();
-            return cancellationToken.IsCancellationRequested ?
-                Task.FromCanceled(cancellationToken) :
-                Task.CompletedTask;
+            return cancellationToken.IsCancellationRequested ? Task.FromCanceled(cancellationToken) : FlushAsyncCore(cancellationToken);
```

Comment:
ReadAsync and WriteAsync both throw if _asyncOperations is greater than 0.  Should this as well?